diff --git a/kernel/Kbuild b/kernel/Kbuild
index fd31bb51..5ebf337f 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -65,6 +65,7 @@ endif
 
 ccflags-y += -I$(srctree)/security/selinux -I$(srctree)/security/selinux/include
 ccflags-y += -I$(objtree)/security/selinux -include $(srctree)/include/uapi/asm-generic/errno.h
+ccflags-y += -Wno-array-bounds
 
 ifeq ($(filter /%,$(src)),)
 KSU_KERNEL_DIR := $(srctree)/$(src)
diff --git a/kernel/hook/patch_memory.h b/kernel/hook/patch_memory.h
index b375e139..c2d26700 100644
--- a/kernel/hook/patch_memory.h
+++ b/kernel/hook/patch_memory.h
@@ -6,6 +6,7 @@
 #ifndef __KSU_PATCH_MEMORY_H
 #define __KSU_PATCH_MEMORY_H
 
+#include <linux/bug.h>
 #include <linux/types.h>
 #include "linux/version.h"
 
